.github/workflows/build.yml

```diff
@@ -19,12 +19,12 @@ jobs:
       - name: apt-get
         run: |
           sudo apt-get update
-          sudo apt-get install libgl1-mesa-dev xorg-dev
+          sudo apt-get install libgl1-mesa-dev xorg-dev mesa-vulkan-drivers
       - uses: actions/setup-go@v3
         with:
           go-version: "1.19"
       - run: go build ./...
-      - run: go test -race ./...
+      - run: go test -short -race ./...
```

Comment:
This might be a better solution: https://peter.bourgon.org/blog/2021/04/02/dont-use-build-tags-for-integration-tests.html

I implemented the same for racedetector tests: https://github.com/Astera-org/obelisk/pull/696/files

We could add an ENV var like `TEST_GPU`, and only run the GPU tests if the flag is set. That means that by default we'd skip the GPU tests (which makes sense for infra & for all people that don't have M1 macs). And it's discoverable, by printing out how to activate the tests.

Reply:
great idea!  note that it should run on any system where the GUI runs -- uses same infrastructure.  there is a ticket for figuring out how to get it to work on software mesa vulkan -- didn't work right away.